R/predict.R

```diff
@@ -521,3 +588,103 @@ rt_by_group <- function(df, group_cols, value_col, vec) {
 
   do.call(rbind, grouped_rt)
 }
+
+#' Manually map day-of-week effects, imputing where possible and outputting
+#' an informative error message where not.
+#' @noRd
+extract_dow_for_predict <- function(object, day_of_week, desired_dates, call) {
+  validate_day_of_week(day_of_week)
+
+  fit_data <- object[["data"]]
+  matching_dates <- desired_dates %in% fit_data[["reference_date"]]
+  all_desired_dates_in_fit <- all(matching_dates)
+  is_custom_vec <- !rlang::is_true(day_of_week) && !rlang::is_false(day_of_week)
+  if (
+    rlang::is_true(day_of_week) && all_desired_dates_in_fit
+  ) {
+    desired_dates <- fit_data[
+      fit_data[["reference_date"]] %in% desired_dates,
+      c("day_of_week", "reference_date")
+    ]
+    # dedupe in case of groups
+    deduped <- unique(desired_dates)
```

Comment:
This assumes there will always be a 1:1 mapping between "reference date" and "day of week". I think that's ok based on your constructor but things could go off the rails if people pass something custom?

Reply:
I don't think this would be possible because the input format enforces a 1:1 map between reference date and day of week and this bit does a lookup on those values.